sdk/core/azure-core/inc/_internal/suffix.hpp

```diff
@@ -0,0 +1,10 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#ifdef _MSC_VER
+#pragma warning(pop)
+#elif defined(__GNUC__) && (__GNUC__ > 4 || (__GNUC__ == 4 && __GNUC_MINOR__ >= 6)) // !_MSC_VER
```

Comment:
Do we really need to do this prefix ans suffix again?
I was hoping to just change from -Wall to -W4
Is there any other lib who comes with -Wall ?

Reply:
What's the use of this prefix.hpp and suffix.hpp?

Reply:
Prefix/suffix is to disable/restore warnings.
For this PR, I am going to follow what's already there; we may change things in the future, but then we'll do it for all files.